core/trino-main/src/main/java/io/trino/metadata/Split.java

```diff
@@ -37,34 +38,51 @@ public record Split(CatalogHandle catalogHandle, ConnectorSplit connectorSplit)
         requireNonNull(connectorSplit, "connectorSplit is null");
     }
 
-    @JsonIgnore
+    @JsonIgnore // TODO remove after https://github.com/airlift/airlift/pull/1141
     public Map<String, String> getInfo()
     {
         return firstNonNull(connectorSplit.getSplitInfo(), ImmutableMap.of());
     }
 
-    @JsonIgnore
+    @JsonIgnore // TODO remove after https://github.com/airlift/airlift/pull/1141
     public List<HostAddress> getAddresses()
     {
         return connectorSplit.getAddresses();
     }
 
-    @JsonIgnore
+    @JsonIgnore // TODO remove after https://github.com/airlift/airlift/pull/1141
     public boolean isRemotelyAccessible()
     {
         return connectorSplit.isRemotelyAccessible();
     }
 
-    @JsonIgnore
+    @JsonIgnore // TODO remove after https://github.com/airlift/airlift/pull/1141
     public SplitWeight getSplitWeight()
     {
         return connectorSplit.getSplitWeight();
     }
 
+    @JsonIgnore // TODO remove after https://github.com/airlift/airlift/pull/1141
     public long getRetainedSizeInBytes()
     {
         return INSTANCE_SIZE
                 + catalogHandle.getRetainedSizeInBytes()
                 + connectorSplit.getRetainedSizeInBytes();
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        // There is no equality contract for ConnectorSplit and there is no requirement for
+        // ConnectorSplits returned from ConnectorSplitSource to be different. For example,
+        // a ConnectorSplitSource could return 10 times same "produce a row" split instance,
+        // so that the table has effectively 10 rows. The blackhole connector does that.
+        return this == o;
```

Comment:
This is not safe to do for objects that can be serialized across the wire. Two split instances that are "equal" to each other according to this method will end up being different when serialized and deserialized.

Reply:
If splits are not expected to be comparable to each other, they should not be records. There's an expectation that records are comparable "by value"

Reply:
I will make https://github.com/trinodb/trino/pull/21631 undraft.